worker/lease/manager.go

```diff
@@ -255,7 +260,14 @@ func (manager *Manager) retryingClaim(claim claim) {
 
 	if lease.IsTimeout(err) {
 		claim.respond(lease.ErrTimeout)
-		manager.config.Logger.Warningf("[%s] retrying timed out while handling claim", manager.logContext)
+		manager.config.Logger.Warningf("[%s] retrying timed out while handling claim %q for %q",
+			manager.logContext, claim.leaseKey, claim.holderName)
+		return
+	} else if lease.IsInvalid(err) {
```

Comment:
If we now have 3 paths for a non-nil error, I would bring up the catacomb.Kill into a switch with these two and poke it inside `if err != nil`. Just a bit less choppy.

Reply:
:+1: